src/libraries/System.Private.CoreLib/src/System/Globalization/IdnMapping.Icu.cs

```diff
@@ -24,15 +24,25 @@ private unsafe string IcuGetAsciiCore(string unicodeString, char* unicode, int c
             if (estimatedLength < StackallocThreshold)
             {
                 char* outputStack = stackalloc char[estimatedLength];
-                actualLength = Interop.Globalization.ToAscii(flags, unicode, count, outputStack, estimatedLength);
+#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
```

Comment:
It would be nice to create a define for this condition - same as what we have in the native code

Reply:
Sure, I agree. I'll create a separate ticket for this, as the use of #if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS is widespread across many .cs files , I prefer not to increase the size of this pull request any further.